### PR TITLE
[WIP] Lexer.

### PR DESCRIPTION
**Pull request by mleconte-06**

### CODE REVIEW

File: srcs/lexer/token_list.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	{";", 1, TOKEN_SEMI},
+	{"\\", 1, TOKEN_BACKSLASH},
+	{"\"", 1, TOKEN_INHIB},
+	{"'", 1, TOKEN_INHIB},
+	{" ", 1, TOKEN_ESP},
+	{"\n", 1, TOKEN_ESP},
+	{"\v", 1, TOKEN_ESP},
+	{"\t", 1, TOKEN_ESP},
+	{"\r", 1, TOKEN_ESP},
+	{"\f", 1, TOKEN_ESP},
```
Comment (jgigault) [inline srcs/lexer/token_list.c]: Perhaps a premature question..
Is `TOKEN_ESP` intended for fields (arguments) separator?
If yes, is that possible to set them dynamically as it should be with $IFS? https://bash.cyberciti.biz/guide/$IFS

File: incs/lexer.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#ifndef TOKEN_H
+# define TOKEN_H
+
+#define TOKEN_BUF_SIZE 256
+
+typedef enum 	e_token_type
+{
+	NOT_A_TOKEN,
```
Comment (jgigault) [inline incs/lexer.h]: ...or `TOKEN_NONE` ? (to keep the same prefix?)

File: srcs/lexer/token_list.c (hunk 1 above)
Comment (mleconte-06) [inline srcs/lexer/token_list.c]: Anouvel noticed me the same.
But for now, it's just a simple lexer implementation, we should talk of that soon.

File: incs/lexer.h (hunk 2 above)
Comment (mleconte-06) [inline incs/lexer.h]: It's okay for me.

**Pull request merged**